rcl/src/rcl/parameter.c

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <rosidl_generator_c/string_functions.h>
+#include <rcl_interfaces/msg/parameter__struct.h>
+#include <rcl_interfaces/msg/parameter_event__struct.h>
+#include <rcl_interfaces/msg/parameter_type__functions.h>
+
+#include "rcl/types.h"
+
+// TODO: Can we assume that other ParameterValue fields are set to NOT_ASSIGNED or should we reassign them?
```

Comment:
It would be a defensive measure to set all the unused fields to NOT_ASSIGNED whenever setting a type field, to enforce that only one type is set in a ParameterValue. Someone could arbitrarily set the fields of struct at any point, which is why I think this is a good idea.